src/storage/src/client.rs

```diff
@@ -426,12 +428,20 @@ impl ReadObject {
         self
     }
 
-    /// Sets the value of [common_object_request_params][control::model::ReadObjectRequest::common_object_request_params].
-    pub fn set_common_object_request_params<T>(mut self, v: T) -> Self
+    /// Sets the value of key in [common_object_request_params][control::model::ReadObjectRequest::common_object_request_params].
+    pub fn set_key<T>(mut self, v: T) -> Self
     where
-        T: Into<control::model::CommonObjectRequestParams>,
+        T: Into<bytes::Bytes>,
```

Comment:
I think you want some specific type because there are length restrictions on that key, no?

See my other comment and let me know if you would rather do that in a future PR.